mountpoint-s3-client/src/endpoint.rs

```diff
@@ -24,13 +22,27 @@ pub struct Endpoint {
 impl Endpoint {
     /// Create a new endpoint for the given S3 region. This method automatically resolves the right
     /// endpoint URI to target.
-    pub fn from_region(region: &str, addressing_style: AddressingStyle) -> Result<Self, EndpointError> {
-        if AWS_PARTITION_REGEX.is_match(region) {
-            // TODO: support partitions other than "aws"
-            Self::from_uri_inner(&format!("https://s3.{region}.amazonaws.com"), addressing_style)
-        } else {
-            Err(EndpointError::UnsupportedRegion(region.to_owned()))
-        }
+    pub fn from_region(
+        region: &str,
+        addressing_style: AddressingStyle,
+        allocator: &Allocator,
+    ) -> Result<Self, EndpointError> {
+        let endpoint_rule_engine = RuleEngine::new(allocator).unwrap();
+        let mut endpoint_request_context = RequestContext::new(allocator).unwrap();
+        endpoint_request_context
+            .add_string(allocator, OsStr::new("Region"), OsStr::new(region))
+            .unwrap();
+        let resolved_endpoint = endpoint_rule_engine
+            .resolve(endpoint_request_context)
+            .map_err(|_| EndpointError::UnsupportedRegion(region.to_owned()))?;
```

Comment:
Since, the only argument I have passed to context is Region, I think Unsupported Region if we are not able to resolve. If we add more parameter to the context, I will add other error types.

Reply:
+1 on this: I do not understand the difference between `UnresolvedEndpoint` and `UnsupportedRegion`

Reply:
I think `UnsupportedRegion` probably just shouldn't exist, unless we feel like we need to distinguish it from other errors somewhere? In general, we're not going to be able to map every possible error from the endpoint resolver -- there are too many.